packages/components/src/components/modal/modal.jsx

```diff
@@ -0,0 +1,122 @@
+import classNames        from 'classnames';
+import PropTypes         from 'prop-types';
+import React             from 'react';
+import ReactDOM          from 'react-dom';
+import { CSSTransition } from 'react-transition-group';
+// import Icon              from 'Assets/icon.jsx';
+
+// TODO: use-from-shared - Use this icon from icons' shared package
+const IconClose = ({ className, onClick }) => (
+    <svg onClick={onClick} className={classNames('inline-icon', className)} width='16' height='16' viewBox='0 0 16 16'>
+        <path className='color1-fill' fill='#2A3052' fillRule='nonzero' d='M8 7.293l4.146-4.147a.5.5 0 0 1 .708.708L8.707 8l4.147 4.146a.5.5 0 0 1-.708.708L8 8.707l-4.146 4.147a.5.5 0 0 1-.708-.708L7.293 8 3.146 3.854a.5.5 0 1 1 .708-.708L8 7.293z' />
+    </svg>
+);
+
+class ModalElement extends React.PureComponent {
+    constructor(props) {
+        super(props);
+        this.el = document.createElement('div');
+        this.state = {
+            modal_root: document.getElementById('modal_root'),
+        };
+    }
+
+    componentDidMount = () => {
+        document.addEventListener('mousedown', this.handleClickOutside);
+        this.el.classList.add('dc-modal');
+        this.state.modal_root.appendChild(this.el);
+    };
+
+    componentWillUnmount = () => {
+        document.removeEventListener('mousedown', this.handleClickOutside);
+        this.state.modal_root.removeChild(this.el);
+    };
+
+    handleClickOutside = (event) => {
+        if (this.wrapper_ref && !this.wrapper_ref.contains(event.target) && this.props.is_open) {
+            this.props.toggleModal();
+        }
+    };
+
+    render() {
+        return ReactDOM.createPortal(
+            <div ref={this.setWrapperRef} id={this.props.id} className={classNames('dc-modal__container', this.props.className && `dc-modal__container_${this.props.className}`)}>
```

Comment:
Can we please put these props on new lines? for all these `div` elements?

Reply:
I agree, it makes detecting changes in future developments easier.

Reply:
@mahdipourziaei-binary @reza-fs its like this every where  else. anyway i don't think you should change the way editors format code much. for instance i even don't know why we align `from` in imports!

Reply:
@Mahboobeh-binary We can't rely on editors as different editors format it differently unless if we have something like Prettier in place. We align `from` in imports as it's easier to visualise when reading.

Reply:
@mahdipourziaei-binary for forms we're currently doing it this way. it was an example.